src/5.0_to_5.1/calculated-measures/migrateCalculatedMeasuresInDashboards.test.ts

```diff
@@ -0,0 +1,155 @@
+import { uiDashboardsFolder } from "../__test_resources__/uiDashboardsFolder";
+import { migrateCalculatedMeasuresInDashboards } from "./migrateCalculatedMeasuresInDashboards";
+import { dataModelsForTests } from "@activeviam/data-model-5.0";
+
+const dataModel = dataModelsForTests.sandbox;
+
+// "pvSum ^ 2" is from cube "EquityDerivativesCubeDist", all others are from "EquityDerivativesCube".
+const calculatedMeasureNames = [
+  "Distinct count city",
+  "Test calculated measure",
+  "EXP pnl.Forex",
+  "Log pv.SUM",
+  "activeui5 calculated measure",
+  "pvSum ^ 2",
+];
+
+describe("migrateCalculatedMeasuresInDashboards", () => {
+  const { migratedDashboards, measureToCubeMapping } =
+    migrateCalculatedMeasuresInDashboards(
+      uiDashboardsFolder,
+      dataModel,
+      calculatedMeasureNames,
+    );
+
+  it("returns an object containing all the calculated measures used in the `ui/dashboards` folder with their corresponding cube name", () => {
+    expect(measureToCubeMapping).toStrictEqual({
+      "Distinct count city": "EquityDerivativesCube",
+      "EXP pnl.Forex": "EquityDerivativesCube",
+      "Log pv.SUM": "EquityDerivativesCube",
+      "Test calculated measure": "EquityDerivativesCube",
+      "activeui5 calculated measure": "EquityDerivativesCube",
+      "pvSum ^ 2": "EquityDerivativesCubeDist",
+    });
+  });
+
+  it("does not modify the content of an empty dashboard", () => {
+    // "8c1" is an empty dashboard.
+    expect(migratedDashboards.children.content.children!["8c1"]).toStrictEqual(
+      uiDashboardsFolder.children.content.children["8c1"],
+    );
+
+    // "a4f" is an empty dashboard inside a folder.
+    expect(migratedDashboards.children.content.children!["a4f"]).toStrictEqual(
+      uiDashboardsFolder.children.content.children["a4f"],
+    );
+  });
+
+  it("does not modify the content of a dashboard which does not contain calculated measures", () => {
+    // "81a" is a dashboard with no calculated measures.
+    expect(migratedDashboards.children.content.children!["81a"]).toStrictEqual(
+      uiDashboardsFolder.children.content.children["81a"],
+    );
+
+    // "e27" is a dashboard with no calculated measures inside a folder.
+    expect(migratedDashboards.children.content.children!["e27"]).toStrictEqual(
+      uiDashboardsFolder.children.content.children["e27"],
+    );
+  });
+
+  it("removes any calculated measure definitions from the `query.mdx` of all widgets from a dashboard with a single page", () => {
```

Comment:
Only the definitions of saved calculated measures are removed, not all of them.
Same everywhere.
```suggestion
  it("removes definitions of calculated measures saved in ActiveUI 5.0 from the `query.mdx` of all widgets from a dashboard with a single page", () => {
```